File: Day-1/MaxDiff.py
def maximumDifference(nums):
    min_value = nums[0]  # First element as the minimum value
    max_difference = -1  # Set max_difference to -1 (indicating no valid pair)

    for j in range(1, len(nums)):
        if nums[j] > min_value:
            # Update with the maximum difference
            max_difference = max(max_difference, nums[j] - min_value)
        # Update with minimum of the current value and nums[j]
        min_value = min(min_value, nums[j])

    return max_difference

File: Day-1/MaxDiff.java
class MaxDiff {
    public int maximumDifference(int[] nums) {
        int minValue = nums[0];  // First element as the minimum value
        int maxDifference = -1;  // Set maxDifference to -1 (indicating no valid pair)

        for (int j = 1; j < nums.length; j++) {
            if (nums[j] > minValue) {
                // Update with the maximum differnce
                maxDifference = Math.max(maxDifference, nums[j] - minValue);
            }
            // Update with minimum of the current value and nums[j]
            minValue = Math.min(minValue, nums[j]);
        }

        return maxDifference;
    }
}